staging/src/k8s.io/apiserver/pkg/server/options/encryptionconfig/config.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package encryptionconfig
+
+import (
+	"crypto/aes"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	yaml "github.com/ghodss/yaml"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/storage/value"
+	aestransformer "k8s.io/apiserver/pkg/storage/value/encrypt/aes"
+	"k8s.io/apiserver/pkg/storage/value/encrypt/identity"
+)
+
+// Config stores the filepath to the encryption provider configuration
+type Config struct {
+	Filepath string
+}
+
+// GetTransformerOverrides returns the transformer overrides by reading and parsing the encryption provider configuration file
+func (config Config) GetTransformerOverrides() (map[schema.GroupResource]value.Transformer, error) {
```

Comment:
Generally we always take functions for things like this, we never add methods.  Your config is a path to a file, so:

```
func ParseTransformerConfig(filepath string) (map[schema.GroupResource]valueTransformer, error) {
```

would be our preferred method.